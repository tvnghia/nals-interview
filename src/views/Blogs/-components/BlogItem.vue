<template>
  <div class="media u-mt-16">
    <img class="media__image" :src="blog.image" alt="blog image" />

    <div class="media__body">
      <p class="headline--md">{{ blog.title }}</p>
      <p class="u-mt-8">{{ blog.content }}</p>
      <p class="headline--sm u-mt-8">Created at: {{ $filters.moment_yyyy_mm_dd(blog.createdAt) }}</p>
    </div>
  </div>
</template>

<script>
export default {
  name: 'BlogItem',

  props: {
    blog: {
      type: Object,
      require: true
    }
  }
}
</script>
