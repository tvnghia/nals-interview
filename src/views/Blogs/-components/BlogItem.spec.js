import { shallowMount } from '@vue/test-utils'
import wrapperOps from '@/config-test/wrapper'
import BlogItem from '@/views/Blogs/-components/BlogItem'

const propsData = {
  blog: {
    content: 'content',
    createdAt: '2020-09-08T12:51:36.252Z',
    id: '1',
    image: 'http://lorempixel.com/640/480/abstract',
    title: 'Oklahoma'
  }
}

Object.assign(wrapperOps, { propsData })

describe('BlogItem.vue', () => {
  const wrapper = shallowMount(BlogItem, wrapperOps)

  describe('when BlogItem was mounted', () => {
    it('should BlogItem render correctly', () => {
      expect(wrapper.exists()).toBeTruthy()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
