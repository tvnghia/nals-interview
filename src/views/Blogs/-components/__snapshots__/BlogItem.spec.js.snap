// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogItem.vue when BlogItem was mounted should BlogItem render correctly 1`] = `
<div class="media u-mt-16"><img src="http://lorempixel.com/640/480/abstract" alt="blog image" class="media__image">
  <div class="media__body">
    <p class="headline--md">Oklahoma</p>
    <p class="u-mt-8">content</p>
    <p class="headline--sm u-mt-8">Created at: 2020/09/08</p>
  </div>
</div>
`;
